app/controllers/players_controller.rb

```diff
@@ -1,13 +1,21 @@
 class PlayersController < ApplicationController
 
   def new
+    @response = Faraday.get("https://graph.facebook.com/v2.10/oauth/access_token?client_id=#{ENV['FACEBOOK_APP_ID']}&redirect_uri=http://localhost:3000/players/new&client_secret=#{ENV['FACEBOOK_APP_SECRET']}&code=#{params["code"]}")
+
+    token = @response.body.split(/\W+/)[2]
+
+    oauth_response = Faraday.get("https://graph.facebook.com/debug_token?input_token=#{token}&access_token=#{ENV['FACEBOOK_APP_ID']}|#{ENV['FACEBOOK_APP_SECRET']}")
+
+    auth = JSON.parse(oauth_response.body)
+    @uid = auth["data"]["user_id"]
     @player = Player.new
   end
 
   def create
     @player = Player.new(player_params)
     if @player.save
-      redirect_to player_profile_path(@player)
+      redirect_to controller: 'player_profile', action: 'new', player_id: @player.id
```

Comment:
I have redirect for recruiter written exactly the same as this, with my routes matching but I'm still getting errors.  I will merge and pull down your changes and compare so I can get recruiter profile creation built.